include/nccl-headers/error.h

```diff
@@ -5,7 +5,7 @@
 #ifndef NCCL_HEADERS_ERROR_H
 #define NCCL_HEADERS_ERROR_H
 
-#if HAVE_CUDA
+#if HAVE_CUDA || HAVE_ROCM
```

Comment:
This is going to get old fast.  Let's just have a HAVE_GPU (assuming that's the name we go with from the previous commit) that is defined in config_bottom.h that is set if either HAVE_CUDA or HAVE_ROCM is set.